### PR TITLE
Add Exception for unexpected case

### PR DESCRIPTION
**Pull request by MrPetovan**
- Address https://github.com/friendica/friendica/issues/10473#issuecomment-882781552
- Address https://github.com/friendica/friendica/issues/10474#issuecomment-882696860

This PR is about whether execution should go forward when `User::getOwnerDataById` doesn't return an owner array. Is there a good reason to proceed when the return is `false`?

### CODE REVIEW

**Comment by annando** [comment]
Have a look here: https://github.com/friendica/friendica/blob/develop/src/Module/Profile/Profile.php#L54-L60

Although the comment confuses me, since it doesn't seem as if the system would return an empty array at all.

**Comment by MrPetovan** [comment]
it isn't much of a problem since we issue a 404 error anyway if the result wasn't a valid array. I'll throw an Exception directly from the method.

**Comment by MrPetovan** [comment]
I've now added the relevant try-catch blocks/documentation for this new Exception throw.

**Comment by annando** [comment]
You are catching the error everywhere - but not in `Probe::localProbe`, why?

**Comment by MrPetovan** [comment]
Because it's the only place where we don't expect it to fail. All the other places have some sort of condition on whether the returned data is empty which needed to be implemented in a `catch` block but not this instance, which means that if an Exception is raised it will expectedly abort the execution of the script.

**Comment by annando** [comment]
Problem is that this function is used in worker tasks as well. When these tasks fail they are repeated - over and over again. This can drastically reduce the worker performance. Also this is used in some processes that are part of the message storing mechanism. So a hard fail here would lead to a loss of posts.

**Comment by MrPetovan** [comment]
I believe we didn't expect this particular call to fail because we perform `User::getIdByUrl` just before and if `$uid` isn't trueish, then `Transmitter::getProfile` doesn't get called at all.

Unlike the other cases where we don't know the provenance of the `uid` parameter, there is a clear dependency chain in this case that doesn't warrant a try-catch block.

**Comment by annando** [comment]
It is called via the `Probe` class when we assume that it is a local contact. It shouldn't fail in any case according to the code. 

Update: I just had a look, `Probe:::uri` is called at nearly 20 different places in the code. This is why I really would like to catch any possible error directly at the source. There **shouldn't** be any problem since we fetch the `uid` value shortly before, but since this happens in all cases I'm unsure how the error could had happened at all.

File: src/Protocol/ActivityPub/Transmitter.php [hunk 1]
```diff
@@ -310,14 +309,20 @@ private static function getService()
...
+	 * @throws HTTPException\NotFoundException
+	 * @throws HTTPException\InternalServerErrorException
 	 */
-	public static function getProfile($uid)
+	public static function getProfile(int $uid): array
 	{
 		$owner = User::getOwnerDataById($uid);
 
+		if (!isset($owner['id'])) {
+			DI::logger()->error('Unable to find owner data for uid', ['uid' => $uid, 'callstack' => System::callstack()]);
```
Comment (annando) [inline src/Protocol/ActivityPub/Transmitter.php]: The `callstack` function only display the last 4 entries by default. When debugging it is helpftul to have some more entries.

Comment (MrPetovan) [reply src/Protocol/ActivityPub/Transmitter.php]: How many do you think I should set it to? Is there an infinite stack parameter value?

**Comment by MrPetovan** [comment]
I'll add a try catch block to the probe call then.

File: src/Protocol/ActivityPub/Transmitter.php (hunk 1 above)
Comment (annando) [reply src/Protocol/ActivityPub/Transmitter.php]: I normally set it to 20.

**Comment by MrPetovan** [comment]
Both the try-catch blocks and the system call stack feedbacks have been addressed.

**Pull request merged**